### ISSUE TITLE
Ubuntu lint checks failing due to lintspaces

### ISSUE THREAD

**Issue by zhaojj2209** [TEAMMATES/teammates#11981]
Lint checks are failing after merging #11910, even though they passed on the PR itself. This can also be replicated locally: lint checks pass when on the branch for #11910 and fail on master.
![image](https://user-images.githubusercontent.com/54243224/198229918-13a7d33d-6034-4d82-852d-3cd6e9bb96da.png)

@tenebrius1 I will be making a PR to revert the changes to lintspaces, in the meantime please help to investigate this further.

**Comment by tenebrius1** [comment]
Hello! I did some investigation, and I am quite certain that this error pops up when the git diff returns nothing i.e., there wasn't any file changed that matched the regex of the git diff. 

A simple fix I could think of is to just append a `|| :`  at the end of the lintspaces command which would basically not run lintspaces if there are no files changed that match the pattern.

New lint:nix:spaces command: `"lint:nix:spaces": "lintspaces -n -t -d spaces -l 1 --matchdotfiles $(git diff --name-only --diff-filter=ACMRTUXB origin/master HEAD | grep -E \"src/(main|web)/.*.html|src/.*.(xml|json|properties)|*.(yml|json|gradle)|static-analysis/*.*ml|.gitattributes\") || :"`

I ran it on the master branch after pulling the latest commit and this is what I got:
![image](https://user-images.githubusercontent.com/57551677/198310361-067a0045-56cc-40d0-acf7-dce73d5f2fa6.png)
It still shows the error that there is no files to lint but fails gracefully afterwards which should not crash the rest of the linting process. Do let me know if this would be an accepted fix!
